06week/higherOrder.js

```diff
@@ -2,26 +2,108 @@
 
 const assert = require('assert');
 
+const test = [1,2,3];
+
+// Create a forEach() function that takes an array of items and a function
+// that runs the function arr.length number of times.
+
 function forEach(arr, callback) {
-  // Your code here
+  for(let i = 0; i < arr.length; i++){
+    if (typeof callback === 'function') {
+      callback(arr[i]);
+    }
+  }
+}
+
+const forEachCallback = () => {
+  return 'forEach: callback is a function';
+}
+
+forEach(test, forEachCallback);
+
+
+// Create a map() function that takes an array of items and a function that
+// returns an array with each item manipulated by that function.
+
+const mapped = [];
+
+function mapCallback(num) {
+  let x = num * num;
+  mapped.push(x);
 }
 
 function map(arr, callback) {
-  // Your code here
+  for (let i = 0; i < arr.length; i++) {
+    callback(arr[i]);
+  }
+  return mapped;
```

Comment:
this would throw an error mapped is not refined. The new array should be stored in the map function